work around bug in firefox which apparently misuses access() as NULL

From: Lennart Poettering <lennart@fefdeb5f-60dc-0310-8127-8f9354f1896f>

pointer test. Original patch by "alon". (Closes #27)

git-svn-id: svn://svn.0pointer.net/pulseaudio/trunk@1352 fefdeb5f-60dc-0310-8127-8f9354f1896f
---

 src/utils/padsp.c |    7 +++++++
 1 files changed, 7 insertions(+), 0 deletions(-)

diff --git a/src/utils/padsp.c b/src/utils/padsp.c
index 32cb5f9..ffe3974 100644
--- a/src/utils/padsp.c
+++ b/src/utils/padsp.c
@@ -2121,6 +2121,13 @@ int close(int fd) {
 }
 
 int access(const char *pathname, int mode) {
+
+    if (!pathname) {
+        /* Firefox needs this. See #27 */
+        errno = EFAULT;
+        return -1;
+    }
+    
     debug(DEBUG_LEVEL_VERBOSE, __FILE__": access(%s)\n", pathname);
 
     if (strcmp(pathname, "/dev/dsp") != 0 &&
